### PR TITLE
docs: update fallback sink docs

### LINKED ISSUE #2302: improve fallback sink doc
We need more details on how to use fallback sink.

1. when to call responseFailure as compared to responseFallback. add it to  https://numaflow.numaproj.io/user-guide/sinks/fallback/
2. how to specify the spec.
* how does `backoff` work with `onFailure` - does it apply to ALL types of onFailure strategies?
add it to https://numaflow.numaproj.io/APIs/#numaflow.numaproj.io/v1alpha1.RetryStrategy
* what are the options for onFailure? 
add it to https://numaflow.numaproj.io/APIs/#numaflow.numaproj.io/v1alpha1.OnFailureRetryStrategy

### PR DESCRIPTION
**Pull request by KeranYang**
Closes #2302

### CODE REVIEW

File: docs/user-guide/sinks/fallback.md
Comment (vigith) [inline docs/user-guide/sinks/fallback.md:65]:
```suggestion
SDKs has to generate a fallback/failed response (e.g., in Java user has to return `responseFallback(String id)`) in the primary user-defined sink:
```

**Review by vigith — APPROVED** [review]

File: docs/user-guide/sinks/fallback.md
Comment (Krithika3) [reply docs/user-guide/sinks/fallback.md:65]: SDKs have to generate a fallback/failed response. (e.g., in Java user has to return `responseFallback(String id)`) in the primary user-defined sink:

**Review by Krithika3 — APPROVED** [review]

**Comment by Krithika3** [comment]
Just for my understanding: let us say we set this
`retryStrategy:
          backoff:
            interval: 500ms
            steps: 4
          onFailure: retry`

what will be the behavior in this case? Infinite retries?

**Comment by KeranYang** [comment]
> Just for my understanding: let us say we set this retryStrategy: backoff: interval: 500ms steps: 4 onFailure: retry what will be the behavior in this case? Infinite retries?

@Krithika3 I didn't look into the code, but based on our current documentation, I believe yes. @kohlisid , could you help confirm?

**Comment by KeranYang** [comment]
@kohlisid , please review to make sure my update are accurate. I didn't look into the implementation. This update is purely based on our discussion this morning.

Folks, please don't merge this one in until Sidhant approves it.

**Review by kohlisid — APPROVED** [review]

**Comment by kohlisid** [comment]
@KeranYang @Krithika3 
Yes, if the strategy is retry it will keep on retrying infinitely.

**Pull request merged**
